front/lib/api/assistant/global_agents.ts

```diff
@@ -825,8 +825,11 @@ export async function getGlobalAgents(
     )
   );
 
-  // ROLLOUT INTERCOM
-  if (!isDevelopmentOrDustWorkspace(owner)) {
+  const isIntercomEnabled = await isFeatureEnabled(
```

Comment:
any chance we can Promise.all this guy with something above ?

Reply:
I added a commit to filter before calling `getGlobalAgent` for intercom, but didn't promise all with something above. Seems weird to promise the intercom feature flag with the fetching of prod credentials or all datasources, no? Happy to do it anyway if you think it's better

Reply:
Not a strong opinion, but IMO if we can fetch stuff concurrently then it tends to be better to reduce latency. really nit and totally fine with the code as-is 👍

Reply:
Refactored to use owner.flags!